seqlock/seqlock.c

```diff
@@ -1,50 +1,46 @@
 #include <assert.h>
+#include <stdatomic.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include "seqlock.h"
 
 #define SEQLOCK_WRITER 1U
 
-#if defined(__i386__) || defined(__x86_64__)
-#define spin_wait() __builtin_ia32_pause()
-#elif defined(__aarch64__)
-#define spin_wait() __asm__ __volatile__("isb\n")
+#if defined(__i386__) || defined(__x86_64__) || defined(__aarch64__)
```

Comment:
Use C11 for non-Aarch64 targets such as x86(-64). For Aarch64, I would stick to `sevl`, `ldx`,  and `wfe` instructions.

Reply:
@jserv I would take your advice and survey some studies about arm's weak ordering issue.